### DISCUSSION TITLE
Tuleap platform support

### DISCUSSION THREAD

**Discussion by wwuck** [renovatebot/renovate#17831] [Suggest an Idea]
I'm looking to migrate off bitbucket server to an alternative platform, and obviously renovate support is a requirement.  I was reviewing [Tuleap](https://www.tuleap.org/) but that doesn't have any renovate support at this time.

How much effort is involved in writing support for a new platform?  Is it just a matter of copying the existing support for another platform and rewriting some methods?  I have looked at the [Tuleap API](https://tuleap.net/api/explorer/) but I'm not sure how to verify if it has all the required endpoints.

**Comment by rarkins** [comment]
The platform API is fairly stable and well documented now. You would probably benefit from looking at the CodeCommit issue for discussions there about adding a new platform recently

**Reply by viceice** [reply]
A quick look at the Tuleap API seems good, should be possible to implement that as renovat platform.
